File: socialuni-js/packages/socialuni-api/src/model/FrontErrorLogVO.ts
import PlatformType from "socialuni-constant/constant/PlatformType";
import RouterUtil from "socialuni-sdk/src/utils/RouterUtil";
import {socialSystemModule} from "socialuni-sdk/src/store/store";

export default class FrontErrorLogVO {
    uri: string = null
    detail: string = null
    params: string = null
    errorMsg: string = null
    platform: string = null
    provider: string = null
    frontPage: string = null
    appVersion: number = null

    constructor(uri: string, detail: string, params: string, errorMsg: string) {
        this.uri = uri
        this.detail = detail
        this.params = params
        this.errorMsg = errorMsg
        this.frontPage = RouterUtil.getCurrentPageURI()
        this.platform = socialSystemModule.platform
        if (this.platform === PlatformType.mp) {
            this.provider = socialSystemModule.mpPlatform
        } else {
            this.appVersion = socialSystemModule.appVersion
        }
    }
}

File: socialuni-server/socialuni-common-sdk/src/main/java/com/socialuni/social/common/sdk/model/VO/FrontErrorLogVO.java
package com.socialuni.social.common.sdk.model.VO;


import lombok.Data;

/**
 * @author qinkaiyuan
 * @date 2018-11-18 20:48
 */
//记录前端错误日志，不该出现的错误
@Data
public class FrontErrorLogVO {
    //出错的url
    private String uri;
    private String frontPage;

    //出错的业务备注
    private String detail;

    //传入了什么参数导致报错
    private String params;

    //错误信息
    private String errorMsg;
    private String platform;
    private String provider;
    private String appVersion;
}
